packages/search/addon/components/navi-search-result/asset.js

```diff
@@ -13,12 +13,14 @@ export default class NaviAssetSearchResultComponent extends Component {
    * @property {Array} results
    */
   get results() {
-    return this.args.data?.map(value => {
-      return {
-        route: this._getRouteFor(value),
-        type: value?.constructor?.modelName,
-        ...value
-      };
+    return this.args?.data?.map(value => {
+      if (!value.route) {
+        value.route = this._getRouteFor(value);
+      }
+      if (!value.type) {
+        value.type = value?.constructor?.modelName;
+      }
```

Comment:
```suggestion
      value.route = value.route || this._getRouteFor(value);
      value.type = value.type || value?.constructor?.modelName;
```

Reply:
👍, but `value?.constructor` -> `value.constructor`

Reply:
I had it this way because its parent object is tracked and if I make changes I get ```Error: Assertion Failed: You attempted to update <dummy@model:report::ember889:8>.route to "reports.report", but it is being tracked by a tracking context, such as a template, computed property, or observer. In order to make sure the context updates properly, you must invalidate the property when updating it. You can mark the property as `@tracked`, or use `@ember/object#set` to do this.``` I'm looking for a better way to tackle it.